src/libraries/Native/Unix/System.Native/pal_runtimeinformation.c

```diff
@@ -52,53 +52,3 @@ int32_t SystemNative_GetUnixVersion(char* version, int* capacity)
 
     return 0;
 }
-
-/* Returns an int representing the OS Architecture:
- 0 - x86
- 1 - x64
- 2 - ARM
- 3 - ARM64
- 4 - WASM */
-int32_t SystemNative_GetOSArchitecture()
-{
-#if defined(TARGET_ARM)
-    return ARCH_ARM;
-#elif defined(TARGET_ARM64)
-    return ARCH_ARM64;
-#elif defined(TARGET_AMD64)
-    return ARCH_X64;
-#elif defined(TARGET_X86)
-    return ARCH_X86;
```

Comment:
When 32-bit dotnet process on linux running on 64-bit OS (in the prescience of libc:i386 / multiarch-support etc.), `RuntimeInformation.OSArchitecture` will return x86 instead of x64. I think we can change it as follows to support this scenario:

```suggestion
#elif defined(TARGET_X86)
    struct utsname _utsname;
    return uname(&_utsname) != -1 && strcmp("x86_64", _utsname.machine) >= 0 ?
        ARCH_X64 : ARCH_X86;
```
(not critical, just noticed this difference in dotnet32 on win64 vs. linux64)

Reply:
This is a bug I'd say, but linux-x86 is not a supported target afaik.

Reply:
linux-x86 is a community-supported platform for a few years, that @t-mustafin and others have been working on from pre-dotnet/runtime days.